docs/dev/settings.rst

```diff
@@ -130,5 +130,43 @@ ELASTICSEARCH_DSL_AUTOSYNC
 
 This setting is used for automatically indexing objects to elasticsearch.
 
-
 .. _elasticsearch-dsl-py.connections.configure: https://elasticsearch-dsl.readthedocs.io/en/stable/configuration.html#multiple-clusters
+
+
+Docker pass-through settings
+----------------------------
+
+If you run a Docker environment, it is possible to pass some secrets through to
+the Docker containers from your host system. For security reasons, we do not
+commit these secrets to our repository. Instead, we individually define these
+settings for our local environments.
+
+We recommend using `direnv`_ for storing local development secrets.
+
+.. _direnv: https://direnv.net/
+
+Allauth secrets
+~~~~~~~~~~~~~~~
+
+It is possible to set the Allauth application secrets for our supported
+providers using the following environment variables:
+
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITHUB_SECRET
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITLAB_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_GITLAB_SECRET
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_OAUTH2_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_OAUTH2_SECRET
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_CLIENT_ID
+.. envvar:: RTD_SOCIALACCOUNT_PROVIDERS_BITBUCKET_SECRET
```

Comment:
These two can be deleted. They are deprecated and we are not using them anymore, even in production.

Reply:
I am going to drop the documentation for these, but `settings.SOCIALACCOUNT_PROVIDERS` will still load these env vars if they exist as it's just looping over the providers defined. I think this is fine for now though, and removing the underlying provider settings will end this.

Reply:
We have an open issue for removing that one from our settings https://github.com/readthedocs/readthedocs.org/issues/11162.